Samples/WindowsCopilotRuntime/cs-wpf-sparse/BuildSparsePackage.ps1

```diff
@@ -0,0 +1,88 @@
+Param(
+    [Parameter(Mandatory=$true)]
+    [string]$Platform = "x64",
+    [Parameter(Mandatory=$true)]
+    [string]$Configuration = "release"
+)
+
+
+function Get-UserPath
+{
+    $root = Join-Path (Get-Item $PSScriptRoot ).FullName "WCRforWPF"
+    $user = Join-Path $root '.user'
+    if (-not(Test-Path -Path $user -PathType Container))
+    {
+        Write-Host "Creating $user..."
+        $null = New-Item -Path $user -ItemType Directory -Force
+    }
+    return $user
+}
+
+$user_path = Get-UserPath
+$pwd_file = Join-Path $user_path 'wcrforwpf.certificate.test.pwd'
+$cert_thumbprint = Join-Path $user_path 'wcrforwpf.certificate.test.thumbprint'
+$cer = Join-Path $user_path 'wcrforwpf.certificate.test.cer'
+$pfx = Join-Path $user_path 'wcrforwpf.certificate.test.pfx'
+
+if (-not (Test-Path $pfx))
+{
+    # clear the files    
+    $files = Get-ChildItem -Path $user_path
+    foreach ($file in $files) {
+        if ($file.PSIsContainer) {
+            Remove-Item -Path $file.FullName -Recurse -Force
+        } else {
+            Remove-Item -Path $file.FullName -Force
+        }
+    }
+    
+    $charSet = 'abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789!@'
+    $passwordLength = 20
+    $password = New-Object -TypeName System.Security.SecureString
+    # Generate random characters and append to SecureString
+    for ($i = 0; $i -lt $passwordLength; $i++)
+    {
+        $randomChar = $charSet[(Get-Random -Maximum $charSet.Length)]
+        $password.AppendChar($randomChar)
+    }
+    
+    # Convert back to plaintext. This is not secure but this cert is only use for development purposes.
+    # Do not use for production purposes.
+    $BSTR = [System.Runtime.InteropServices.Marshal]::SecureStringToBSTR($password)
+    $passwordPlainText = [System.Runtime.InteropServices.Marshal]::PtrToStringAuto($BSTR)
+    [Runtime.InteropServices.Marshal]::ZeroFreeBSTR($BSTR)
+    
+    Set-Content -Path $pwd_file -Value $passwordPlainText -Force
+    
+    # Create the certificate
+    $cert_path = "cert:\CurrentUser\My"
+    $now = Get-Date
+    $expiration = $now.AddMonths(12)
+    $subject = "CN=Microsoft Corporation, O=Microsoft Corporation, L=Redmond, S=Washington, C=US"
+    $friendly_name = "Microsoft.WindowsAppSDK Test Certificate Create=$now"
+    $key_friendly_name = "Microsoft.WindowsAppSDK Test PrivateKey Create=$now"
+    $key_description = "Microsoft.WindowsAppSDK Test PrivateKey Create=$now"
+    $eku_oid = '2.5.29.37'
+    $eku_value = '1.3.6.1.5.5.7.3.3,1.3.6.1.4.1.311.10.3.13'
```

Comment:
What are these magic values, and are these appropriate for use?

Reply:
Howard said verbatim these black magic values should be untouched